0002-data-collection-extension/README.md

```diff
@@ -1,111 +1,124 @@
+- **Data sample**: an independent data record collected from an endpoint.
+- **Batch**: a collection of data samples uploaded in a single request message.
 - **Processing confirmation**: an acknowledgment designating that the server has finished processing a request.
 
+
 # Requirements and constraints
-Data Collection protocol requirements:
 
-- Record processing is asynchronous.
-Processing might require a significant period of time and may be performed simultaneously or in a different order.
+- Data sample processing is asynchronous.
+Processing data samples might require a significant period of time and may be performed simultaneously or in a different order than sent.
 
-- The client must know if a record has been processed.
+- The client may need to know if a data sample has been processed.
 This is needed for the client to guarantee that the data has been processed and to free allocated resources.
 
   Possible solutions:
   - Use QoS levels for message delivery confirmation.
   - Use QoS levels for message processing confirmation.
     MQTT requires all PUBACK messages appear in the same order as received PUBLISH messages.
-    This creates a synchronization point, which conflicts with asynchronous record processing requirement.
+    This creates a synchronization point, which conflicts with asynchronous processing requirement.
   - Use separate response messages for processing confirmation.
-
     This is achieved by request/response pattern as defined per 1/KP.
 
-- The server should handle different types of data.
+- The server should be able to handle different data formats.
   Different endpoints may send data in a variety of formats. The server should know the format to parse the payload.
 
   Solutions:
-  - Use `<format_specifier>` embedded into resource path.
+  - Use `<format_specifier>` embedded into the resource path.
 
 - The server should know the endpoint that generates the data.
 
   Solutions:
-  - Make 2/DCX an endpoint-aware extension as defined per 1/KP.
+  - Make 2/DCP an endpoint-aware extension as defined per 1/KP.
 
-- Little network usage.
+- Minimize network usage.
   Internet connection may be slow, unstable, or costly, so the extension should send as little data as possible.
 
   Solutions:
-  - Upload data in batches.
-    A batch is a number of records uploaded in one network packet.
+  - Upload data in [batches](#Language).
 
-- Device can be unable to generate timestamps.
+- Device may not be able to generate timestamps.
 
   Solutions:
   - On the server side, use network packet arrival time as a timestamp if no timestamp is present.
 
+
 # Use cases
+
+
 ## UC1: Device shadow
+
 The user only wants to know the current status of the endpoint parameters.
 The endpoint updates them periodically.
 
+
 ## UC2: Historical data collection
+
 The user wants to store all collected data with timestamps for further processing and for visualizing historical trends.
 
+
 # Design
+
+
```

Comment:
I suggest 2 empty lines before every heading and one after for better raw markdown readability.